addons/l10n_tr_nilvera_einvoice_extended/models/account_move.py

```diff
@@ -0,0 +1,99 @@
+        string="Invoice Scenario",
+        help="The scenario of the invoice to be sent to GİB.",
+    )
+
+    l10n_tr_gib_invoice_type = fields.Selection(
+        compute="_compute_l10n_tr_gib_invoice_type",
+        store=True,
+        readonly=False,
+        default="SATIS",
+        selection=[
+            ("SATIS", "Sales"),
+            ("TEVKIFAT", "Withholding"),
+            ("IHRACKAYITLI", "Registered for Export"),
+            ("ISTISNA", "Tax Exempt"),
+        ],
+        help="The type of sales invoice to be sent to GİB.",
+    )
+
+    l10n_tr_is_export_invoice = fields.Boolean(string="Is GIB Export")
+
+    l10n_tr_shipping_type = fields.Selection(
+        selection=[
+            ("1", "Sea Transportation"),
+            ("2", "Railway Transportation"),
+            ("3", "Road Transport"),
+            ("4", "Air Transportation"),
+            ("5", "Post"),
+            ("6", "Combined Transportation"),
+            ("7", "Fixed Transport"),
+            ("8", "Domestic Water Transportation"),
+            ("9", "Invalid Transport Method"),
+        ],
+        string="Shipping Method",
+        help="The type of shipping.",
+    )
+    l10n_tr_exemption_code_id = fields.Many2one(
+        "l10n_tr.account.tax.code",
+        compute="_compute_l10n_tr_exemption_code_id",
+        store=True,
+        readonly=False,
+        string="Exemption Reason",
+        domain=lambda self: str(self._l10n_tr_exemption_code_domain()),
+    )
+
+    l10n_tr_exemption_code_domain_list = fields.Binary(
+        compute="_compute_l10n_tr_exemption_code_domain_list"
+    )
+
+    l10n_tr_nilvera_customer_status = fields.Selection(
+        string="Partner Nilvera Status",
+        related="partner_id.l10n_tr_nilvera_customer_status",
+    )
+
+    def _l10n_tr_exemption_code_domain(self):
+        return [
+            ("code_type", "in", unquote("l10n_tr_exemption_code_domain_list") or []),
+        ]
+
+    @api.depends("l10n_tr_gib_invoice_scenario", "l10n_tr_gib_invoice_type")
+    def _compute_l10n_tr_exemption_code_domain_list(self):
+        for record in self:
+            domain = []
+            if record.l10n_tr_gib_invoice_type == "ISTISNA":
+                domain.append("exception")
+                domain.append("export_exception")
+            if record.l10n_tr_gib_invoice_type == "IHRACKAYITLI":
+                domain.append("export_registration")
+            if record.l10n_tr_is_export_invoice:
+                domain.append("export_exception")
+            record.l10n_tr_exemption_code_domain_list = domain
+
+    @api.depends("l10n_tr_gib_invoice_scenario", "l10n_tr_is_export_invoice")
+    def _compute_l10n_tr_gib_invoice_type(self):
+        self.l10n_tr_gib_invoice_type = False
+
+    @api.depends(
+        "l10n_tr_gib_invoice_scenario", "l10n_tr_gib_invoice_type", "partner_id"
+    )
+    def _compute_l10n_tr_exemption_code_id(self):
+        self.l10n_tr_exemption_code_id = False
```

Comment:
I'm not so sure about this compute with the default... Yes, the value is set to the default when creating a move, but a move could be created with any value through the code, so you could have a l10n_tr_gib_invoice_scenario different from its default, and still have the l10n_tr_gib_invoice_type to its default. Would that be ok ?

I'm wondering if we should not just remove the default on l10n_tr_exemption_code_id and let the user decide, or really automate the process by changing the compute to actually set some value when needed